ballista/rust/core/src/serde/mod.rs

```diff
@@ -508,15 +508,10 @@ mod tests {
             &self,
             children: Vec<Arc<dyn ExecutionPlan>>,
         ) -> datafusion::error::Result<Arc<dyn ExecutionPlan>> {
-            match children.len() {
-                1 => Ok(Arc::new(TopKExec {
-                    input: children[0].clone(),
-                    k: self.k,
-                })),
-                _ => Err(DataFusionError::Internal(
-                    "TopKExec wrong number of children".to_string(),
-                )),
-            }
+            Ok(Arc::new(TopKExec {
```

Comment:
I removed the length check in the  `with_new_children()` method and move the check to `with_new_children_if_necessary()` so that we can avoid the duplicate logic in all the trait implementations. In future if someone misuse this and still call the `with_new_children()` and pass the wrong children param, it will lose the check, but if this is the case,  the real problem is he should not call `with_new_children()` and everyone should call `with_new_children_if_necessary()`.

Reply:
I'm not sure in Rust whether I can declare a method in the Trait to `protected` so that it can only be visible in this module and in the Trait implementations.